packages/eslint-plugin/src/rules/no-extra-parens.ts

```diff
@@ -256,6 +241,56 @@ export default util.createRule<Options, MessageIds>({
         }
       },
     };
+    if (semver.satisfies(TSESLint.ESLint.version, '>=7.19.0')) {
```

Comment:
Instead of using a semver check here, as these can be pretty hairy to get right.

Just guard appropriately.

```js
if (rules.ForInStatement && rules.ForOfStatement) {
  // new version
} else {
  // old version
}
```

Reply:
Nice, thanks!